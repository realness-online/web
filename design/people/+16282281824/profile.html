<figure itemscope itemtype="/person" itemid='/people/+16282281824'>
  <time itemprop="created_at" datetime="2018-05-10T16:17:39.624Z"></time>
  <time itemprop="updated_at" datetime="2018-05-10T16:17:39.624Z"></time>
  <svg>
    <use itemprop="profile_vector" xlink:href="/static/icons.svg#silhouette"></use>
  </svg>
  <figcaption>
    <p><span itemprop="first_name">Scott</span><span itemprop="last_name">Fryxell</span></p>
    <a itemprop="mobile" data-value="+16282281824"></a>
  </figcaption>
</figure>



<figure itemscope itemtype="/person" itemid='/people/+16282281824'>
  <svg><use itemprop="profile_vector" xlink:href="/static/icons.svg#silhouette"></use></svg>
  <figcaption>
    <p><span itemprop="first_name">Scott</span><span itemprop="last_name">Fryxell</span></p>
    <a itemprop="mobile" data-value="+16282281824"></a>
  </figcaption>
</figure>
