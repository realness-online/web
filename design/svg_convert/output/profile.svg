<svg version="1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 128 128"><path d="M37.8 7.9c-3.1 2-6.1 4.4-6.8 5.4-.9 1.5-.9 1.8.2 1.2.7-.5-.4 1-2.6 3.3-2.5 2.7-3.7 4.6-3.2 5.7.3.9.2 1.4-.3 1.1-1.5-1-4.1 8.2-5.6 19.4-2.1 15-3.8 24-4.7 24-.4 0-.8.8-.8 1.7-.1 1.8-2.2 5.3-3.3 5.3-1.3 0-5.7 5.2-5.2 6.1.6.9 0 1.1-3.5 1.3-1.3.1-1.5 3.5-1.8 22.9L-.1 128h35.9c30.7 0 35.9-.2 35.9-1.5 0-.8-.4-1.2-.9-.9-.4.3-.9-.5-.9-1.8-.1-1.3-.2-3-.3-3.8-.1-.8.3-2.1.7-2.7 1.3-1.8.1-4.3-2.2-4.3-2 0-3.1-1.8-3.1-5 0-1.4 1.7-2.4 6.8-4.1 6.7-2.2 7.4-3.9.9-2.1-2.7.8-3.8.6-4.9-.5-.8-.8-2.4-1.5-3.6-1.6-1.9-.1-2.1-.6-1.7-4.1.3-2.2 1-4.3 1.7-4.7 1.6-1 .2-6.7-2.2-8.7-2.2-1.9-2.6-5.7-1.5-13.7.9-6.4-.9-15.7-3.6-18.1-1.1-1-3-1.9-4.2-2.1-1.2-.2-3.9-.8-6-1.4-3.7-1-3.7-1-3.6-6 .1-2.7.6-5.2 1.1-5.5 1.4-.9.9-5.2-.9-6.9-1-1-1.2-1.9-.6-2.3.5-.4 1.3-2 1.7-3.5.4-1.5 1.6-3.1 2.9-3.6 1.2-.5 4.6-2.3 7.5-4 5.9-3.4 6.9-3.6 14.3-2 3.4.8 6.4 2.3 9 4.6 2.2 2 3.9 3.2 3.9 2.7 0-.4 1.2.3 2.6 1.7l2.7 2.4-.7-3c-.4-1.7-.9-3.8-1.3-4.9-.4-1.3-.2-1.6.8-1 2.5 1.5-3.3-5.6-6.4-8-4.6-3.4-7.1-3.9-11.3-2.1-2.2.9-5 1.3-6.6 1-2.6-.6-3.3 0-3 2.6.2.9-5.8 2-6.9 1.3-.5-.3-.9-1-.9-1.5 0-.6.5-.7 1-.4.6.3 1.5.3 2.1 0 1.7-1.1-2.2-3.2-6.7-3.6-3.2-.3-5.1.2-9.6 3zM47.5 54c.3.5-.7 1.3-2.2 1.7l-2.8.7 3 .8c2.2.5 1.7.6-1.5.2-5.3-.6-5.7-.9-4-2.9 1.4-1.7 6.5-2 7.5-.5z"/><path d="M82.6 22c.4.8 1.3 2.2 2 3 .7.8 2.4 3.7 3.9 6.5 1.4 2.7 2.3 3.9 1.9 2.5-1.2-4.5-2.3-6.5-5.4-10-1.8-1.9-2.8-2.8-2.4-2zM74.6 42.7c-1.2 1.2-.6 2.5 1 1.9.8-.3 1.2-.2.9.4-.3.6.1 1 .9 1 .9 0 1.6-.4 1.6-1 0-1.2-3.5-3.1-4.4-2.3zM72.1 50.1c-1.7 1-3.1 2.4-3.1 3.1 0 2.3 1.7 4.1 3 3.3.8-.5 1.1-.4.6.3-.7 1.1 4.6 4.2 7.3 4.2 1.4 0 1.4-.2.1-1-1.2-.8-1.1-1 .6-1 1.1 0 2.6-.6 3.3-1.3.6-.6 2.4-1.2 3.8-1.2s2.7-.6 2.9-1.3c.1-.7 0-.8-.2-.4-.5 1-7.2-1-8.3-2.4-.3-.5-2-1.6-3.8-2.5-2.9-1.6-3.3-1.6-6.2.2zm1.4 1.9c.3.5-.1 1-1 1s-1.3-.5-1-1c.3-.6.8-1 1-1 .2 0 .7.4 1 1zM94.3 55.5c0 2.2.2 3 .4 1.7.2-1.2.2-3 0-4-.3-.9-.5.1-.4 2.3zM100.6 75.7c-.4 3.8-.9 9.5-1.1 12.8-.8 10.3-3.1 23.9-4.7 27-.9 1.9-1.3 4.7-1 7.7.4 4.2.7 4.8 2.7 4.8 2.1 0 2.4-.5 2.3-3.9 0-2.2.3-4.5.7-5.2.5-.7.4-1-.2-.6-1.6 1-2.3-4.2-1.3-8.8 1.6-7.4 3-18.9 3.6-29.8.7-11.8 0-14.6-1-4zM69 83c0 .5.7 1 1.5 1s1.5-.5 1.5-1c0-.6-.7-1-1.5-1s-1.5.4-1.5 1zM90.1 123.5c-2.3 4-2.5 4.5-1.3 4.5 1 0 4.5-7 3.9-7.7-.3-.2-1.4 1.2-2.6 3.2zM123.7 125.7c-1.4 1.3-.6 2.3 1.9 2.3 2.2 0 2.5-.2 1.4-1.5-1.4-1.6-2.3-1.9-3.3-.8z"/></svg>