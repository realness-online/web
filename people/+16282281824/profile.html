<figure id="profile" itemscope="itemscope" itemtype="/person">
  <time itemprop="created_at" datetime="2018-05-10T16:17:39.624Z"></time>
  <time itemprop="updated_at" datetime="2018-05-10T16:17:39.624Z"></time>
  <svg >
    <use itemprop="profile_vector" xlink:href="/static/icons.svg#silhouette"></use>
  </svg>
  <figcaption>
    <p><span itemprop="first_name">Scott</span><span itemprop="last_name">Fryxell</span></p>
    <a itemprop="mobile" data-value="+16282281824"></a>
  </figcaption>
</figure>
