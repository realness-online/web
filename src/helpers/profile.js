import itemid from '@/helpers/itemid'
import * as firebase from 'firebase/app'
import 'firebase/storage'
export default {
  async load (id) {
    const person = await itemid.load(`${id}/index`)
    if (person) return person
    else return { id }
  },
  async directory (id, type) {
    const storage = firebase.storage().ref()
    const path = `/people/${id}/${type}`
    if (navigator.onLine) {
      return storage.child(path).listAll()
    } else return null
  },
  as_query_id (id = '/+') {
    return id.substring(2)
  },
  as_avatar_id (id = 'avatar_') {
    return `avatar_${this.as_query_id(id)}`
  },
  as_avatar_fragment (id = 'avatar_') {
    return `#${this.as_avatar_id(id)}`
  },
  as_fragment (id) {
    return `#${this.as_query_id(id)}`
  },
  as_phone_number (id = '/+1') {
    return id.substring(3)
  },
  from_phone_number (phone_number) {
    return `/+1${phone_number}`
  },
  from_e64 (e64_number) {
    return `/${e64_number}`
  }
}
