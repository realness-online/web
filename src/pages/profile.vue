<template lang="html">
  <section id="profile" class="page" v-bind:class="{me}">
    <header>
      <profile-as-figure :person='person' :me="me" :previous="true"></profile-as-figure>
      <logo-as-link></logo-as-link>
    </header>
    <icon v-show="working" name="working"></icon>
    <my-posts v-if="me"></my-posts>
    <posts-list v-else :posts='posts'></posts-list>
  </section>
</template>
<script>
  import '@/modules/timeago'
  import {person_storage} from '@/modules/Storage'
  import profileAsFigure from '@/components/profile/as-figure'
  import logoAsLink from '@/components/logo-as-link'
  import postsList from '@/components/posts/as-list'
  import myPosts from '@/components/posts/my-list'
  import icon from '@/components/icon'
  import load_mobile_item from '@/mixins/load_mobile_item'
  export default {
    mixins: [load_mobile_item],
    components: {
      profileAsFigure,
      myPosts,
      postsList,
      logoAsLink,
      icon
    },
    created() {
      const mobile = this.$route.params.mobile
      if (mobile) {
        this.get_items_from_mobile(mobile, 'person').then(items => {
          this.working = false
          this.person = items[0]
        })
        this.get_items_from_mobile(mobile, 'posts').then(items => {
          this.posts = items
        })
      } else {
        this.me = true
        this.working = false
        this.person = person_storage.as_object()
      }
    },
    data() {
      return {
        me: false,
        working: true,
        person: {},
        posts: {}
      }
    }
  }
</script>
<style lang='stylus'>
  @require '../style/variables'
  section#profile > svg.working
    margin-top: base-line
</style>
