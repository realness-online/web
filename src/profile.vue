<template lang="html">
  <section id="profile">
    <header>
      <h1>Profile</h1>
    </header>
  </section>
</template>

<style lang='stylus'>
  @require 'style/variables'
  section#profile
    padding: 0 base-line
    & > details
      margin: base-line 0
</style>
