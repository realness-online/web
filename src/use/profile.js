export function as_phone_number(id = '/+1') {
  return id.substring(3)
}
export function from_e64(e64_number) {
  return `/${e64_number}`
}
