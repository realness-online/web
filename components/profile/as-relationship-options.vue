<template>
  <a
    class="status"
    :class="{ relation: is_relation }"
    @click="update_relationship">
    <icon name="add" />
    <icon name="finished" />
  </a>
</template>
<script setup>
  import Icon from '@/components/icon'
  import { use_me } from '@/use/people'
  import { nextTick as next_tick, computed } from 'vue'
  import { Relations } from '@/persistance/Storage'
  const props = defineProps({
    person: {
      type: Object,
      required: true
    }
  })
  const { relations } = use_me()
  const is_relation = computed(() => {
    return relations.value.some(relation => relation.id === props.person.id)
  })
  const update_relationship = async () => {
    if (is_relation.value) {
      const index = relations.value.findIndex(p => p.id === props.person.id)
      if (index > -1) relations.value.splice(index, 1)
      if (!relations.value.length)
        localStorage.removeItem(`${localStorage.me}/relations`)
    } else relations.value.push(props.person)
    await next_tick()
    await new Relations().save()
  }
</script>
<style lang="stylus">
  figure.profile > figcaption > menu > a.status
    svg
      transition: fill, width, height
      transition-duration: 0.5s
      cursor: pointer
      fill: blue
      width:  base-line
      height: base-line
      &.finished
        fill: green
        height: 0
        width: 0
    &.relation
      svg.add
        height: 0
        width: 0
      svg.finished
        width:  base-line
        height: base-line
</style>
