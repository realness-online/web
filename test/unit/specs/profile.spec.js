import Vue from 'vue'
import { shallow } from 'vue-test-utils'
import profile from '@/profile'

describe('index.vue', () => {
  it('render layour for who you know', () => {
    let wrapper = shallow(profile)
    expect(wrapper.element).toMatchSnapshot()
  })
})
