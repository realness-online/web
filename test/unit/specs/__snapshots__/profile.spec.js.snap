// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index.vue render layour for who you know 1`] = `
<section
  id="profile"
>
  <header>
    <h1>
      Profile
    </h1>
  </header>
</section>
`;
